src/with-actions.ts

```diff
@@ -8,8 +8,8 @@ type InferStateActions<Actions> = Actions extends {
     }
   : unknown;
 
-// FIXME we should check name collisions between state and actions (help wanted)
-type IsValidActions<_State, _Actions> = true;
+type IsValidActions<State, Actions> =
+  Extract<keyof State, keyof Actions> extends never ? true : false;
```

Comment:
Okay, it was easy. It was hard with `withSlices` because of arrays. Thanks for your work!

Reply:
No problem, looking forward to collaborating further! Absolutely, `withSlices` was more challenging.

Reply:
Great. What's missing in zustand-slices, maybe docs?

Reply:
Let's create `docs` directory. Feel free to work on anything you thing is good. Maybe I can help with the motivation section.

Reply:
Great, thank you! I will start working on organizing the `docs` directory and integrating the suggested enhancements.